.github/workflows/native_build_check_android.yml

```diff
@@ -0,0 +1,15 @@
+name: '[Code Quality] Check Native Android Builds'
+description: 'Tests to see that changes to native files do not break the Android build process'
+
+on:
+  pull_request:
+    paths:
+      - "VAMobile/android/**"
+
+jobs:
+  android_native_build:
+    name: Android Test Build
+    uses: ./.github/workflows/build_android.yml
+    secrets: inherit
+    with:
+      lane: test_build_only
```

Comment:
These two checks are on the actual building of app .ipa or .aab file to make sure there are no issues if we update the native files. updating those files could change the build settings for the final artifacts and cause errors that bundle wont catch

Reply:
yeah I'm hesitant to include fastlane checks in a PR since there are other reasons that can fail including random timeouts.

Reply:
these checks do not happen on a regular PR, only when something in the actual native files changes. 

any changes outside of those files just gets the bundle check along with the tests.

Reply:
from the attached ticket 

> There are instances where we may want to confirm that changes in a PR didn't break the app build. This would serve as a gate to prevent changes that don't require manual QA from being merged into develop without confirming the build still works. This is mainly a concern when updating the folders for native code, e.g. VAMobile/android, VAMobile/ios.

Reply:
> other reasons that can fail including random timeouts.

You have seen timeouts on the actual build step, or on the uploading/verifying upload steps after the build?

Reply:
We see various instances of fastlane failing. Usually on the upload after the build. If we're not actually uploading, I suppose that's fine

Reply:
there shouldnt be timouts on builds, and since we are just building the artifact and not uploading you should not see timeouts. 

With these in place you now have unit tests, e2e tests, bundle tests (that test whether the js transpiles from ts correctly), and these other build tests (that make sure that pods/gradle/native changes dont break the final artifacts )

that covers a huge portion of edge cases of broken stuff